internal/formatter/formatter.go

```diff
@@ -0,0 +1,87 @@
+package formatter
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// Format method formats files inside of the integration directory.
+func Format(packageRoot string, failFast bool) error {
+	err := filepath.Walk(packageRoot, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+		err = formatFile(path, failFast)
+		if err != nil {
+			return errors.Wrapf(err, "formatting file failed (path: %s)", path)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "walking through the integration files failed")
+	}
+	return nil
+}
+
+func formatFile(path string, failFast bool) error {
+	file := filepath.Base(path)
+	ext := filepath.Ext(file)
+
+	var c interface{}
+	var unmarshal func([]byte, interface{}) error
+	var marshal func(interface{}, string, string) ([]byte, error)
+	switch ext {
```

Comment:
I wonder if we should use https://golang.org/pkg/net/http/#DetectContentType instead of relying on the extension, especially since you are opening the file further down in this function anyway.

Reply:
Take a look at the internals of this method. It depends on mime types definition of the operating system, not really the Go code. I think it last detection of JSON and YAML content type.

Reply:
Hmmm, looks like `http.DetectContentType` cannot detect JSON or YAML after all. So ++ to relying on the extension instead.